### ISSUE TITLE
[DEV] failed to create directory '/install/kvm117': File exists

### ISSUE THREAD

**Issue by penguhyang** [xcat2/xcat-core#721]
arch: ppc64le (x86_64 works fine)
MN: c910f02c09p16 VMhost1:c910f05c33 VMhost2:c910f05c35
VM: kvm117
how to procedure
MN:
```
mkdir /install/kvm117
vi /etc/exports #add kvm117
exportfs -r
```
VMhost1: c910f05c33
```
mkdir /install/kvm117
mount c910f02c09p16:/install/kvm117 /install/kvm117
```
MN:
```
mkdef -t node kvm117 arch=ppc64le groups=kvm,vm,all mgt=kvm netboot=grub2 profile=compute serialport=0 serialspeed=115200 vmcpus=2 vmhost=c910f05c33 vmmemory=1024 vmnicnicmodel=virtio vmnics=br0 vmstorage=dir:///install/kvm117
mkvm kvm117 -s 20G
```
err msg:
```
[root@c910f02c09p16 install]# mkvm kvm117 -s 20G
kvm117: Error: 38, message: failed to create directory '/install/kvm117': File exists: Unknown issue libvirt error code
```

**Comment by daniceexi** [comment]
@immarvin Is this the same error you encountered?

**Comment by whowutwut** [comment]
I'd like to work on this one..

**Comment by whowutwut** [comment]
@penguhyang  is this on xCAT 2.11 or master branch?

**Comment by penguhyang** [comment]
@whowutwut Thanks for review. Yes, it's on master branch.

**Comment by gurevichmark** [comment]
The problem appears to be not nfs specific. It seems that when running mkvm, the setting of vmstorage is used
to create a new directory. So in the scenario above, if vmstorage=dir:///install/kvm117/myowndir and myowndir does not exist, the command will work and will create a subdir **myowndir** and store qcow2 file there.

Similarly, if vmstorage=dir:///var/lib/libvirt/images/mgimags is specified and a normal non nfs directory mgimages exists there, the mkvm will also fail, just like in the nfs scenario.

Investigating why the setting vmstorage=dir:///var/lib/libvirt/images containing multiple qcow2 files does not complain that directory is already there.

**Comment by gurevichmark** [comment]
This issue is fixed by pull request #1331

**Comment by penguhyang** [comment]
Leave a comment
